src/qiboconnection/api.py

```diff
@@ -367,6 +367,8 @@ def execute(
         if not selected_devices:
             raise ValueError("No devices were selected for execution.")
 
+        if not isinstance(circuit, list):
+            circuit = [circuit]
```

Comment:
Please add a test for this!! 🙏

Reply:
was struggling with that because no there's no unit test done for execute and I'm not very good at mocking yet. As a bypass, I will encapsulate this in a separate function to move quickly.